docs/tools/plist.md

```diff
@@ -0,0 +1,301 @@
+# Static Analysis PLIST Format Specification
+
+Plist files are used in CodeChecker to store and parse diagnostic 
+information from static analysis tools.
+This documentation explains the structure of plist files generated by 
+CodeChecker, illustrating how they are represented in the code and what 
+they look like on the CodeChecker UI.
+
+## Table of Contents
+* [Plist files in CodeChecker](#plist-files-in-codechecker)
+* [Structure of a .plist File](#structure-of-a-plist-file)
+* [Diagnostic Entry Examples](#diagnostic-entry-examples)
+  * [1. `bug_path_event` — Bug Path Events](#1-bug_path_event--bug-path-events)
+  * [2. `notes` — Notes](#2-notes--notes)
+  * [3. `macro_expansions` — Macro Expansions](#3-macro_expansions--macro-expansions)
+  * [4. `report-annotation` — Report Annotations](#4-report-annotation--report-annotations)
+* [License](#license)
+
+
+## Plist files in CodeChecker
+
+Plist files are produced by the report-converter in CodeChecker from the 
+analyzer outputs. They are used to store and parse diagnostic information from 
+static analysis tools. For usage, parameters, and supported analyzers of 
+the report-converter, refer to the relevant documentation file.
+
+## Structure of a .plist File
+
+### Example Structure
+
+```xml
+<plist version="1.0">
+<dict>
+	<key>diagnostics</key>
+	<array>
+		<dict>
+			<key>category</key>
+			<string>unknown</string>
+			<key>check_name</key>
+			<string>division-by-zero</string>
+			<key>description</key>
+			<string>The message of the report object is shown here</string>
+			<key>issue_hash_content_of_line_in_context</key>
+			<string>...</string>
+			<key>location</key>
+			<dict>
+				<key>col</key>
+				<integer>42</integer>
+				<key>file</key>
+				<integer>0</integer>
+				<key>line</key>
+				<integer>8</integer>
+			</dict>
+			<key>path</key>
+			<array>
+				...
+			</array>
+			<key>type</key>
+			<string>report-converter-type</string>
+		</dict>
+	</array>
+	<key>files</key>
+	<array>
+		<string>files/file.cpp</string>
+	</array>
+	<key>metadata</key>
+	<dict>
+		<key>analyzer</key>
+		<dict>
+			<key>name</key>
+			<string>...</string>
+		</dict>
+		<key>generated_by</key>
+		<dict>
+			<key>name</key>
+			<string>report-converter</string>
+			<key>version</key>
+			<string>x.y.z</string>
+		</dict>
+	</dict>
+</dict>
+</plist>
+```
+
```

Comment:
Here please explain the meaning of each key.
* `category:string` The category of the checker emitted the report. Currently unused by the CodeChecker tooling
* `check_name:string` The name of the checker emitted the report. This checker name is visualized in the GUI and used in the report suppressions.
* `description:string` The short description of the analysis report.
...